src/Cli/Extras/Process.hs

```diff
@@ -244,14 +252,20 @@ streamToLog stream = fix $ \loop -> do
 waitForProcess :: MonadIO m => ProcessHandle -> m ExitCode
 waitForProcess = liftIO . Process.waitForProcess
 
+-- | Converts an 'ExitCode' to an exception when it's non-zero.
```

Comment:
Not too convinced about the naming/documentation here: It doesn't _convert_ an `ExitCode` to an exception, it _throws_ an exception on `ExitFailure`. Can we change the name/doc to better reflect this?